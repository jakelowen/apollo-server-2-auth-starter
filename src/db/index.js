const environment = process.env.NODE_ENV || 'development';
const knexTinyLogger = require('knex-tiny-logger').default;
const config = require('../../knexfile')[environment];

// eslint-disable-next-line import/order
const db = require('knex')(config);

const { DEBUG } = process.env;
if (DEBUG) knexTinyLogger(db);

module.exports = db;
