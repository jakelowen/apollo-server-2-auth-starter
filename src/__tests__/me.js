/* eslint-disable no-undef */
import gql from 'graphql-tag';
import { createTestClient } from 'apollo-server-testing';
import { constructTestServer, makeTestContext } from './__utils';
import db from '../db';

require('dotenv').config();

const ME_QUERY = gql`
    {
      me {
        id
        name
        email
        roles
      }
    }
`;

beforeAll(async () => db.migrate.latest());
beforeEach(async () => Promise.all([db.raw('TRUNCATE TABLE users CASCADE')]));
afterAll(async () => db.destroy());

describe('ME', () => {
  test('Happy Path no perms', async () => {
    const email = 'fake124@test.com';
    const id = '0b986756-e6e3-4bec-bb3c-fe5ce28d0752';
    const name = 'fake dude';
    await db.table('users').insert({
      id, name, email,
    });

    const context = await makeTestContext({ req: { userId: id } }); // user: { id, email, name } }),
    // console.log('CONTEXT user', context.user);
    const { server } = constructTestServer({
      context,
    });

    const { query } = createTestClient(server);
    const res = await query({
      mutation: ME_QUERY,
    });
    expect(res).toMatchSnapshot();
  });

  test('Happy Path with perms', async () => {
    const email = 'fake124@test.com';
    const id = '0b986756-e6e3-4bec-bb3c-fe5ce28d0752';
    const name = 'fake dude';
    await db.table('users').insert({
      id, name, email,
    });

    await db('permission').insert([
      { user_id: id, role: 'foo' },
      { user_id: id, role: 'bar' },
    ]);

    const context = await makeTestContext({ req: { userId: id } }); // user: { id, email, name } }),
    // console.log('CONTEXT user', context.user);
    const { server } = constructTestServer({
      context,
    });

    const { query } = createTestClient(server);
    const res = await query({
      mutation: ME_QUERY,
    });
    expect(res).toMatchSnapshot();
  });


  test('throws error if not logged in', async () => {
    const clearCookie = jest.fn();
    const { server } = constructTestServer({
      context: await makeTestContext({ res: { clearCookie } }),
    });
    const { query } = createTestClient(server);
    const res = await query({ mutation: ME_QUERY });
    expect(res).toMatchSnapshot();
  });
});
