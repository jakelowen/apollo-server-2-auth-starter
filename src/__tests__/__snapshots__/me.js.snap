// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ME Happy Path no perms 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "fake124@test.com",
      "id": "0b986756-e6e3-4bec-bb3c-fe5ce28d0752",
      "name": "fake dude",
      "roles": Array [
        "applicant",
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`ME Happy Path with perms 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "fake124@test.com",
      "id": "0b986756-e6e3-4bec-bb3c-fe5ce28d0752",
      "name": "fake dude",
      "roles": Array [
        "bar",
        "foo",
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`ME throws error if not logged in 1`] = `
Object {
  "data": Object {
    "me": null,
  },
  "errors": Array [
    [GraphQLError: You must be logged in to do this],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
